arangod/IResearch/IResearchLinkCoordinator.cpp

```diff
@@ -140,6 +140,44 @@ Result IResearchLinkCoordinator::init(velocypack::Slice definition) {
 
   return r;
 }
+IResearchDataStore::Stats IResearchLinkCoordinator::stats() const {
+  auto& cmf = IResearchLink::collection()
+                  .vocbase()
+                  .server()
+                  .getFeature<metrics::ClusterMetricsFeature>();
+  auto data = cmf.getData();
+  if (!data) {
+    return {};
+  }
+  auto& metrics = data->metrics;
+  std::string labels;
+  auto addLabel = [&](std::string_view key, std::string_view value) {
+    if (!labels.empty()) {
+      labels.push_back(',');
+    }
+    labels.append(key);
+    labels.push_back('=');
+    labels.push_back('"');
+    labels.append(value);
+    labels.push_back('"');
+  };
+  addLabel("db", getDbName());
+  addLabel("view", getViewId());
+  addLabel("collection", getCollectionName());
+  auto getValue = [&](const std::string& key) {
+    if (auto it = metrics.find({key, labels}); it != metrics.end()) {
```

Comment:
it seems we do a lot of allocations here, can we improve it anyhow?

Reply:
I think I should use something like 
`absl::flat_hash_map<std::string_view, absl::flat_hash_map<std::string, Value>>`
Also I think we can make labels once in ctor?
In that case no allocation